### DISCUSSION TITLE
Help with using `TABBY_CONFIG_DIRECTORY` environment variable.

### DISCUSSION THREAD

**Discussion by timkinnane** [Eugeny/tabby#9523] [Q&A]
Hi I'm trying to follow the steps described in [this issue comment](https://github.com/Eugeny/tabby/pull/9048#issuecomment-1877804923) to add a Tabby config.yaml to my Mac dotfiles, like `$HOME/.tabby/config.yaml`.

I would like to follow first option to "set this environment variable on the operating system level according to your specific operating system".

I have tried exporting the variable in my user shell profile `~/.zshrc` like `export TABBY_CONFIG_DIRECTORY=$HOME/.tabby` but it isn't getting picked up by the Tabby client, it still loads config from the default path `$HOME/Library/Application Support/tabby/config.yaml`. I also tried setting it in `~/.profile`, `~/.bashrc` and `~/.bash_profile`.

I'd like to avoid using the `.env` file in the Tabby install path, because I'm trying to get a setup I can reproduce on concurrent installs, i.e. something static I can commit to my dotfiles repo and pull down on new machines or after formatting.

Here's the config reading code for reference: https://github.com/Eugeny/tabby/blob/master/app/lib/config.ts

**Comment by Eugeny** [comment]
IIRC profile files only apply to interactive shells. Try this https://stackoverflow.com/questions/135688/setting-environment-variables-on-os-x , but first confirm it works the way you expect it to by directly running `TABBY_CONFIG_DIRECTORY=$HOME/.tabby /Applications/Tabby.app/Contents/MacOS/Tabby`

**Reply by timkinnane** [reply]
Thanks Eugeny. I tried a couple of approaches from that article:
- Adding `setenv TABBY_CONFIG_DIRECTORY ~/tabby` to `/etc/launchd.conf`
- Calling `launchctl setenv TABBY_CONFIG_DIRECTORY ~/tabby-test`

Those both kind of work (had some strange behaviour but I don't know if it's related). In any case they aren't the kind of solution I was looking for re static definitions I can commit to a dotfiles repo.

I ended up using [stow](https://www.gnu.org/s/stow/manual/stow.html) to symlink the default config path into my dotfiles:

`stow -v -t "$HOME/Library/Application Support" -d $DOTFILES -S tabby`

Thanks for the support.
